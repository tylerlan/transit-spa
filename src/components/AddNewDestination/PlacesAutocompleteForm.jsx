import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PlacesAutocomplete from 'react-places-autocomplete';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { addDestination } from '../../actions';
import './button.css';

export class PlacesAutocompleteForm extends Component {
  constructor(props) {
    super(props);
    this.state = { address: '600 Guerrero St, San Francisco, CA 94110' };
    this.onChange = address => this.setState({ address });
  }

  handleFormSubmit = (event) => {
    event.preventDefault();
    this.props.onClick();
    this.props.addDestination(this.state.address);
  };

  render() {
    const inputProps = {
      value: this.state.address,
      onChange: this.onChange,
      placeholder: 'ZOOM!',
    };

    return (
      <form className="ui semantic" onSubmit={this.handleFormSubmit}>
        <PlacesAutocomplete inputProps={inputProps} />
        <button className="ui green button" id="submit-destination" type="submit">
          Submit
        </button>
      </form>
    );
  }
}

PlacesAutocompleteForm.propTypes = {
  onClick: PropTypes.func.isRequired,
  addDestination: PropTypes.func.isRequired,
};

PlacesAutocompleteForm.defaultProps = {
  onClick: () => {},
  addDestination: () => {},
};

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      addDestination,
    },
    dispatch,
  );

export default connect(null, mapDispatchToProps)(PlacesAutocompleteForm);
