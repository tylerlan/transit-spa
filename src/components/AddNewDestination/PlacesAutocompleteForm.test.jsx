import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PlacesAutocompleteForm } from './PlacesAutocompleteForm';

describe('PlacesAutocompleteForm', () => {
  it('should render an empty form', () => {
    const component = shallow(<PlacesAutocompleteForm />);
    expect(toJson(component)).toMatchSnapshot();
  });
});

/*
There is a submit button in the form but it's not possible to test it. The
reason is that when mounting the component a Google library is loaded and it
needs to have a window context.
*/
