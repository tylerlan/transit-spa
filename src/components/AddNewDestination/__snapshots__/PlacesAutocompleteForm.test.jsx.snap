// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesAutocompleteForm should render an empty form 1`] = `
<form
  className="ui semantic"
  onSubmit={[Function]}
>
  <PlacesAutocomplete
    autocompleteItem={[Function]}
    classNames={Object {}}
    clearItemsOnError={false}
    debounce={200}
    highlightFirstSuggestion={false}
    inputProps={
      Object {
        "onChange": [Function],
        "placeholder": "ZOOM!",
        "value": "600 Guerrero St, San Francisco, CA 94110",
      }
    }
    onError={[Function]}
    options={Object {}}
    styles={Object {}}
  />
  <button
    className="ui green button"
    id="submit-destination"
    type="submit"
  >
    Submit
  </button>
</form>
`;
